table/block_based/partitioned_filter_block.cc

```diff
@@ -76,10 +76,13 @@ void PartitionedFilterBlockBuilder::MaybeCutAFilterBlock(
   // Add the prefix of the next key before finishing the partition. This hack,
```

Comment:
Suggest

    ... before finishing the partition, without updating last_prefix_str.